tests/unittests/benchmark/task/profet/test_profet_task.py

```diff
@@ -0,0 +1,375 @@
+        self,
+        profet_train_config: MetaModelTrainingConfig,
+        profet_input_dir: Path,
+        tmp_path_factory,
+    ):
+        """ TODO: Test that when instantiating multiple tasks with the same arguments, the
+        meta-model is trained only once.
+        """
+        max_trials = 123
+        task_id = 0
+        seed = 123
+
+        checkpoint_dir: Path = tmp_path_factory.mktemp("checkpoints")
+        # Directory should be empty.
+        assert len(list(checkpoint_dir.iterdir())) == 0
+
+        task = self.Task(
+            max_trials=max_trials,
+            task_id=task_id,
+            train_config=profet_train_config,
+            seed=seed,
+            input_dir=profet_input_dir,
+            checkpoint_dir=checkpoint_dir,
+        )
+        assert task.max_trials == max_trials
+        assert task.seed == seed
+        assert task.task_id == task_id
+
+        # Directory should have one file (the trained model).
+        assert len(list(checkpoint_dir.iterdir())) == 1
+
+        task = self.Task(
+            max_trials=max_trials,
+            task_id=task_id,
+            train_config=profet_train_config,
+            seed=seed,
+            input_dir=profet_input_dir,
+            checkpoint_dir=checkpoint_dir,
+        )
+        assert task.max_trials == max_trials
+        assert task.seed == seed
+        assert task.task_id == task_id
+
+        # Directory should have one file (the trained model).
+        assert len(list(checkpoint_dir.iterdir())) == 1
+
+    def test_configuration(
+        self,
+        profet_train_config: MetaModelTrainingConfig,
+        profet_input_dir: Path,
+        checkpoint_dir: Path,
+    ):
+        """ Test that tasks have a proper configuration and that they can be created from it.
+        """
+        task_id = 0
+        kwargs = dict(
+            max_trials=10,
+            task_id=task_id,
+            train_config=profet_train_config,
+            seed=123,
+            input_dir=profet_input_dir,
+            checkpoint_dir=checkpoint_dir,
+        )
+        first_task = self.Task(
+            **kwargs
+        )
+        configuration = first_task.configuration
+        assert len(configuration.keys()) == 1
+        key, config_dict = configuration.popitem()
+        assert key == self.Task.__qualname__
+        
+        for key, value in kwargs.items():
+            assert key in config_dict
+            assert config_dict[key] == value
+
+        # NOTE: This 'name' is the qualname of the class, which is used by the Factory metaclass'
+        # `__call__` method.
+        name = self.Task.__qualname__
+        from orion.benchmark.benchmark_client import _get_task
+        second_task = _get_task(name=name, **config_dict)
+        assert second_task.configuration == first_task.configuration        
+        
+
+    @pytest.mark.xfail(
+        reason="TODO: Not sure how to modify the `space.sample` for a particular space instance "
```

Comment:
I do not think we should try doing that for now. We should rely on `space_object.sample(seed)` to stay coherent with the rest of the code.